incubator/validator-json/src/main/java/io/aklivity/zilla/runtime/validator/JsonValidator.java

```diff
@@ -0,0 +1,157 @@
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.function.LongFunction;
+import java.util.function.ToLongFunction;
+import java.util.stream.Collectors;
+
+import jakarta.json.spi.JsonProvider;
+import jakarta.json.stream.JsonParser;
+import jakarta.json.stream.JsonParserFactory;
+
+import org.agrona.DirectBuffer;
+import org.agrona.collections.Long2ObjectHashMap;
+import org.leadpony.justify.api.JsonSchema;
+import org.leadpony.justify.api.JsonSchemaReader;
+import org.leadpony.justify.api.JsonValidationService;
+import org.leadpony.justify.api.ProblemHandler;
+
+import io.aklivity.zilla.runtime.engine.catalog.CatalogHandler;
+import io.aklivity.zilla.runtime.engine.config.CatalogedConfig;
+import io.aklivity.zilla.runtime.engine.config.SchemaConfig;
+import io.aklivity.zilla.runtime.engine.validator.Validator;
+import io.aklivity.zilla.runtime.validator.config.JsonValidatorConfig;
+
+public class JsonValidator implements Validator
+{
+    private static final byte MAGIC_BYTE = 0x0;
+
+    private final JsonProvider schemaProvider;
+    private final Long2ObjectHashMap<CatalogHandler> handlersById;
+    private final JsonValidationService service;
+    private final JsonParserFactory factory;
+    private final List<CatalogedConfig> catalogs;
+    private final SchemaConfig catalog;
+    private final CatalogHandler handler;
+
+    public JsonValidator(
+        JsonValidatorConfig config,
+        ToLongFunction<String> resolveId,
+        LongFunction<CatalogHandler> supplyCatalog)
+    {
+        this.handlersById = new Long2ObjectHashMap<>();
+        this.schemaProvider = JsonProvider.provider();
+        this.service = JsonValidationService.newInstance();
+        this.factory = schemaProvider.createParserFactory(null);
+        this.catalogs = config.catalogs.stream().map(c ->
+        {
+            c.id = resolveId.applyAsLong(c.name);
+            handlersById.put(c.id, supplyCatalog.apply(c.id));
+            return c;
+        }).collect(Collectors.toList());
+        this.catalog = catalogs.get(0).schemas.size() != 0 ? catalogs.get(0).schemas.get(0) : null;
+        this.handler = handlersById.get(catalogs.get(0).id);
+    }
+
+    @Override
+    public boolean read(
+        DirectBuffer data,
+        int index,
+        int length)
+    {
+        boolean status = false;
+        byte[] payloadBytes = new byte[length];
+        data.getBytes(0, payloadBytes);
+        ByteBuffer byteBuf = ByteBuffer.wrap(payloadBytes);
+
+        if (byteBuf.get() == MAGIC_BYTE)
+        {
+            int schemaId = byteBuf.getInt();
+            int valLength = length - 1 - 4;
+            byte[] valBytes = new byte[valLength];
+            data.getBytes(length - valLength, valBytes);
+
+            String schema = handler.resolve(schemaId);
+
+            if (schema != null && validate(schema, valBytes))
+            {
+                status = true;
+            }
+        }
+        return status;
```

Comment:
We need to be able to validate `json` without the magic byte as we're going to use this generically to validate `http` request content, or `mqtt` message payload, not just in `kafka` encoded messages.